tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/Package/VersionUpdateTool.cs

```diff
@@ -0,0 +1,255 @@
+            {
+                releaseDate = DateTime.Now.ToString("yyyy-MM-dd");
+                logger.LogInformation("No release date specified. Setting to current date: {ReleaseDate}", releaseDate);
+            }
+            else
+            {
+                // Validate date format (YYYY-MM-DD)
+                if (!DateTime.TryParseExact(releaseDate, "yyyy-MM-dd", null, System.Globalization.DateTimeStyles.None, out _))
+                {
+                    return PackageOperationResponse.CreateFailure(
+                        $"Invalid release date format: {releaseDate}. Expected format: YYYY-MM-DD (e.g., 2025-11-12)",
+                        nextSteps: [
+                            "Provide the release date in the correct format: YYYY-MM-DD",
+                            "Re-run the tool with the corrected release date"
+                        ]);
+                }
+            }
+
+            // Set default release type to 'beta' if both Version and ReleaseType are empty or null
+            if (string.IsNullOrWhiteSpace(version) && string.IsNullOrWhiteSpace(releaseType))
+            {
+                releaseType = "beta";
+                logger.LogInformation("No version or release type specified. Defaulting to release type: beta");
+            }
+
+            // Discover the repository root
+            var sdkRepoRoot = gitHelper.DiscoverRepoRoot(packagePath);
+            if (sdkRepoRoot == null)
+            {
+                return PackageOperationResponse.CreateFailure("Unable to find git repository root from the provided package path.");
+            }
+
+            logger.LogInformation("Repository root discovered: {SdkRepoRoot}", sdkRepoRoot);
+            var languageService = GetLanguageService(packagePath);
+            if (languageService == null)
+            {
+                return PackageOperationResponse.CreateFailure("Tooling error: unable to determine language service for the specified package path.", nextSteps: ["Create an issue at the https://github.com/Azure/azure-sdk-tools/issues/new", "contact the Azure SDK team for assistance."]);
+            }
+
+            // Check for package type
+            var packageInfo = await languageService.GetPackageInfo(packagePath, ct);
+            if (packageInfo?.SdkType == SdkType.Management)
+            {
+                // For management-plane packages, execute configured version update script
+                var (configContentType, configValue) = await _specGenSdkConfigHelper.GetConfigurationAsync(sdkRepoRoot, SpecGenSdkConfigType.UpdateVersion);
+                if (configContentType != SpecGenSdkConfigContentType.Unknown && !string.IsNullOrEmpty(configValue))
+                {
+                    logger.LogInformation("Found valid configuration for updating version. Executing configured script...");
+
+                    // Prepare script parameters
+                    var scriptParameters = new Dictionary<string, string>
```

Comment:
@haolingdong-msft @haiyuazhang @msyyc @MaryGao @tadelesh @skywing918 
this is the place to prepare the parameters for the language script call.